src/webgpu/api/validation/copyTextureToTexture.spec.ts

```diff
@@ -210,3 +236,222 @@ g.test('multisampled_copy_restrictions')
       isSuccess
     );
   });
+
+g.test('uncompressed_texture_format_equality')
+  .params(
+    params()
+      .combine(poptions('srcFormat', kUncompressedTextureFormats))
+      .combine(poptions('dstFormat', kUncompressedTextureFormats))
+  )
+  .fn(async t => {
+    const { srcFormat, dstFormat } = t.params;
+
+    const kTextureSize = { width: 16, height: 16, depth: 1 };
+
+    const srcTexture = t.device.createTexture({
+      size: kTextureSize,
+      format: srcFormat,
+      usage: GPUTextureUsage.COPY_SRC,
+    });
+
+    const dstTexture = t.device.createTexture({
+      size: kTextureSize,
+      format: dstFormat,
+      usage: GPUTextureUsage.COPY_DST,
+    });
+
+    const isSuccess = srcFormat === dstFormat;
+    t.TestCopyTextureToTexture(
+      { texture: srcTexture },
+      { texture: dstTexture },
+      kTextureSize,
+      isSuccess
+    );
+  });
+
+g.test('depth_stencil_copy_restrictions')
+  .params(
+    params()
+      .combine(poptions('format', kDepthStencilFormats))
+      .combine(
```

Comment:
The parametrization seems a bit much maybe. What we want to test can be split for src and dst (or just have a parameter that inverts them). And for example for src, we want to test for a couple different levels with:

 - a full size copy for the miplevel
 - copies that are like fullsize but with just an offset of 1 in top, bottom, left and right.

So maybe we could have the following:

```
  poptions('copyRectangleOffsets' [
    {x:0, y:0, width:0, height:0},
    {x:1, y:0, width:0, height:0},
    {x:0, y:1, width:0, height:0},
    {x:0, y:0, width:-1, height:0},
    {x:0, y:0, width:0, height:-1},
  ])
```

Reply:
I'm not sure I have fully understood your comments here. PTAL, thanks!

Reply:
The offsets stuff looks good. Let's not worry about the src vs. dst part.

Reply:
src vs dst is already done by just testing both directions in the test function, which lgtm